lib/WALASupport/SILWalaInstructionVisitor.cpp

```diff
@@ -721,88 +721,52 @@ jobject SILWalaInstructionVisitor::visitCondBranchInst(CondBranchInst *CBI) {
 }
 
 jobject SILWalaInstructionVisitor::visitSwitchValueInst(SwitchValueInst *SVI) {
-
   SILValue Cond = SVI->getOperand();
-  jobject CondNode = findAndRemoveCAstNode(Cond.getOpaqueValue()); 
+  jobject CondNode = findAndRemoveCAstNode(Cond.getOpaqueValue());
   
   if (Print) {
     llvm::outs() << "\t [COND]: " << Cond.getOpaqueValue() << "\n";
   }
 
-  SILBasicBlock *FalseBasicBlock = nullptr;
-  SILBasicBlock *TrueBasicBlock = nullptr;
-  
-  // Evaluate cases
+  // Make children
+  list<jobject> Children;
+
   for (unsigned Idx = 0, Num = SVI->getNumCases(); Idx < Num; ++Idx) {
     auto Case = SVI->getCase(Idx);
     auto *CaseVal = dyn_cast<IntegerLiteralInst>(Case.first);
+    SILBasicBlock *CaseBasicBlock = Case.second;
 
-    if (!CaseVal)
-      return nullptr;
-    SILBasicBlock *DestBasicBlock = Case.second;
-    if (CaseVal->getValue() == 0) {
-      FalseBasicBlock = DestBasicBlock;
-    } else {
-      TrueBasicBlock = DestBasicBlock;
-    }
-  }
-
-  // Check for defaults
-  if (SVI->hasDefault()) {
-    if (!FalseBasicBlock) {
-      FalseBasicBlock = SVI->getDefaultBB();
-    } else if (!TrueBasicBlock) {
-      TrueBasicBlock = SVI->getDefaultBB();
-    }
-  }
+    Children.push_back(Wala->makeConstant(CaseVal));
```

Comment:
`CaseVal` should have been handled. Instead of creating a new constant, we should use `findAndRemoveCAstNode` to find the correspond `jobject`.